components/gfx/display_list/mod.rs

```diff
@@ -309,14 +302,16 @@ impl StackingContext {
             };
 
             if opts::get().dump_display_list_optimized {
-                println!("**** optimized display list. Tile bounds: {:?}", tile_bounds);
+                println!("**** optimized display list. Tile bounds: {:?}", paint_context.page_rect);
                 display_list.print_items("*".to_owned());
             }
 
             // Sort positioned children according to z-index.
             let mut positioned_children: SmallVec<[Arc<StackingContext>; 8]> = SmallVec::new();
             for kid in display_list.children.iter() {
-                positioned_children.push((*kid).clone());
+                if kid.layer.is_none() {
+                    positioned_children.push((*kid).clone());
+                }
```

Comment:
It's a slight optimization - the if checks below previously excluded them anyway, this just avoids adding them to the smallvec at all.